### ISSUE TITLE
Block/Mute/Filter individual post/toot without interacting with it

### ISSUE THREAD

**Issue by mc776** [mastodon/mastodon#21583]
### Pitch

If someone posts something annoying, or something that was originally good but I've come to find annoying after the 17th time it's been boosted by a mutual (and a different one each time), or if it results in an ongoing argument involving multiple people I follow and I couldn't care less about that argument, I want to be able to filter out that post and all its replies and never see it again without navigating specifically to it.

### Motivation

Feature already exists in Pleroma and it's one big thing I always miss when switching from Akkoma to Mastodon.

When I run into the situation described in the pitch, in Akkoma I can just mute the thread, forgive this beloved mutual for their indiscretion and move on with my life.

In Mastodon my only options are:
- block the person who was boosted;
- unfollow/mute/block my own follow;
- scroll past it every single time it's boosted or replied to by someone I follow, cope, seethe, eventually break down and do one of the above two;
- end up so angry and fed up with the post that I make a #FediBlock post about it.

I've blocked so many people on Mastodon before getting my first non-Mastodon alt and realizing what was going on, and it's completely unnecessary.

**Comment by afontenot** [comment]
Maybe I don't fully understand what you want, but tell me if this works for you.

1. Go to the filters section of your settings. Click to make a new filter.
2. Name it something like Filtered Posts. Set it to apply to all contexts. Add a nonsense word like ybyrKdXqpokEVCsLoqAhKhF as the only keyword (something that won't match anything). Accept the new filter.
3. Now, while viewing a post you don't want to see any more, go to the menu, click "filter this message". Add the post to the "filtered posts" filter.

Now you shouldn't see the post any more.

Honestly this feels like a very strange way to have to do this but it does seem to work. The idea of "filtering" posts makes less sense to me than muting / hiding them.

**Comment by mc776** [comment]
...huh.

Okay that does give the intended end result, but it seems extremely counterintuitive. The filter selection screen doesn't explain anything and I thought it was an option to add more keywords to a given filter instead.

It's also a lot of clicks with the confirmation screen and everything, for something that by definition we want to spend a minimum amount of time ever thinking about again.

Why not have a separate filtered posts list that everything gets dumped into?

**Comment by mc776** [comment]
My ideal setup:

- the filter settings in the preferences lets you select which filter the individual posts will get dumped into.
- when you click "Filter this post", a little confirmation popup appears on the bottom left saying "Post added to <name of filter>". No other feedback except the post is no longer on the screen.

**Comment by afontenot** [comment]
Yep I totally agree that the current implementation is counter-intuitive. I don't think of muting posts as a "filtering" procedure at all. I'd rather have a "mute post" button and (optionally) a list of posts that I've muted under the settings. But this does solve the problem, at least for now?

Related: https://github.com/mastodon/mastodon/issues/20711 https://github.com/mastodon/mastodon/issues/5470 https://github.com/mastodon/mastodon/issues/1669

**Comment by mc776** [comment]
20,000 statuses and a total change of vocabulary later...

**Comment by jumase** [comment]
> Now you shouldn't see the post any more.

That's true, you don't see the post anymore. However, replies are displayed. As I understood, the idea is that the original toot **and all replies** should be hidden, isn't it?

> 1. Go to the filters section of your settings. Click to make a new filter.
> 
> 2. Name it something like Filtered Posts. Set it to apply to all contexts. Add a nonsense word like ybyrKdXqpokEVCsLoqAhKhF as the only keyword (something that won't match anything). Accept the new filter.
> 
> 3. Now, while viewing a post you don't want to see any more, go to the menu, click "filter this message". Add the post to the "filtered posts" filter.

By the way, it's not necessary to previously create a filter (steps 1 and 2). You can directly create the specific filter right away from the menu you pointed out. Then you can edit time span and so on.

So to sumarize I think we need **an option to also hide/silence replies to that toot**, even from other followers.

**Comment by mc776** [comment]
Just tried it in a real live situation, can filter the individual post but it doesn't take away the replies.

**Comment by eishiya** [comment]
I've run into a few cases where I've wanted to be able to mute a thread *with replies*, and also found it surprising that filtering a post doesn't filter the replies. Some examples:
- A thread of funny screencaps from other social media, people reblog their favourite ones. I don't want to see this, and I'd like to mute the thread without muting the user, or hiding all boosts.
- A periodically updated thread about a followed user's project I'm not interested in. Some people use unique per-thread tags people can mute, but that clutters their posts and they have to remember to use those tags.
- Threads where two people I follow are just having a non-DM chat I'm not involved in and don't care about. Because I follow both of them, their posts mentioning each other show up on my TL (which is normally great!), but it would be nice to mute when the conversations get long or irrelevant to me.

**Comment by salviex** [comment]
Please re-evaluate this. I need an option to hide threads (original post and every reply and retoot to it).

I follow a bunch of nice people. Sometimes, out of boredom, one of them starts a toot game: "Rock song titles, but with a gardening twist" (you know the sort of threads). Then it goes back and forth, filling up my entire feed. As if that weren’t enough, the next morning I get retoots from another acquaintance, which push it all back into my feed again. 
And I really don't want to unfollow or block these folks. 

I'm therefore requesting a feature:
- Hide an entire thread
- with all replies
- with all retoots from that thread
